internal/cache/controller_obj_cache.go

```diff
@@ -3320,8 +3318,8 @@ func validateAndConfigureSeGroup(client *clients.AviClient, returnErr *error) bo
 		if aviError, ok := err.(session.AviError); ok && aviError.HttpStatusCode == 403 {
 			//SE in provider context no read access
 			utils.AviLog.Debugf("Switching to admin context from  %s", lib.GetTenant())
-			SetAdminTenant(client.AviSession)
-			defer SetTenant(client.AviSession)
+			//fallback to Admin Tenant
+			client = SharedAVIClients(lib.GetAdminTenant()).AviClient[0]
```

Comment:
@abhishekbsingh: Here we should create new variable like `client := Shared..`  else original variable will be replaced.

Reply:
With this change, we are creating additional 8 client (more memory consumed) than just switching tenant for one operation without creating any new variables.

Reply:
8 clients for admin tenant will anyway be created during PopulateCache method. Also, client replacement should be ok as the client is being passed to ConfigureSeGroupLabels as well.

Reply:
@abhishekbsingh If Tenant is non-admin, then populateCache will have 8 clients for that tenant. So here 8 more clients will be created.

Reply:
@abhishekbsingh: Also `validateAndConfigureSeGroup` is being called from `ValidateUserInput`. So with this change, client will be replaced for all subsequent function calls (instead of using actual tenant, admin tenant will be used.)